### PR TITLE
Added the Thumper

### PR DESCRIPTION
**Pull request by Mailaender**
now that we have sandworms https://github.com/OpenRA/OpenRA/pull/6821 and merged medic and engineer into a new support infantry.

### CODE REVIEW

**Comment by penev92** [comment]
OK, #7278 is merged, so there's nothing more stopping this.

**Comment by Mailaender** [comment]
I don't really know what you want me to do now. Should the thumper spawn sandworms or attract them (to get eaten) or not attract them at all?

**Comment by penev92** [comment]
I was thinking attract them. Probably should get a really big Noise value in order to do its job properly.

**Comment by Phrohdoh** [comment]
Sorry I don't know much about Dune (other than BFD). Does this unit deploy? If so I have an idea.

**Comment by penev92** [comment]
I'd say with the current sound system you should give the thumper noise Intensity of 1000-1400 and override the default noise Falloff with something like `{ 100, 50 }` on a 
`WRange Spread = new WRange(27648);` (27 cells).
That is just pure guessing and will still need to be tested ingame and balanced.

**Comment by Mailaender** [comment]
Rebased this to experiment with the worm changes. The thumper won't spawn, but attract worms now. I have not redone the transformation from scratch as demanded by @pchote.

**Comment by abcdefg30** [comment]
Found a weird bug. Steps to reproduce:
Get a thumper and a rifle man. Hit the thumper two times.
thumpera
Deploy the thumper, undeploy it and then redeploy it.
Enjoy your invulnerable and not undeployable* thumper!
thumperb
(*means you can't undeploy it.)

**Comment by Mailaender** [comment]
Does this also work with MCVs?

**Comment by abcdefg30** [comment]
I could reproduce it in ra.
But the mcv needs to have almost no health.

**Comment by Mailaender** [comment]
Can you make a ticket about that exploit?

File: mods/d2k/rules/infantry.yaml [hunk 1]
```diff
@@ -47,64 +47,111 @@ ENGINEER:
...
+	LoopedSound:
+		TickDelay: 18
+		Filename: THUMPER1.WAV
+	WithBuildingExplosion:
+	AttractsWorms:
+		Intensity: 1400
+		Falloff: 100, 50
+		Spread: 27648
+
+
```
Comment (abcdefg30) [inline mods/d2k/rules/infantry.yaml]: Doubled new line.

**Comment by pchote** [comment]
You can get rid of the transformation by making `AttractsWorms` an `UpgradableTrait` and using an upgrade to prevent movement and enable to a higher-noise attraction.

**Comment by abcdefg30** [comment]
I got this as I deployed the thumper before a worm attack:
thumpereaten

**Comment by Mailaender** [comment]
That is probably a duplicate of https://github.com/OpenRA/OpenRA/issues/4754.

File: mods/d2k/rules/infantry.yaml (hunk 1 above)
Comment (Mailaender) [inline mods/d2k/rules/infantry.yaml]: Fixed.

**Comment by Mailaender** [comment]
> You can get rid of the transformation by making `AttractsWorms` an `UpgradableTrait` and using an upgrade to prevent movement and enable a higher-noise attraction.

That won't change the artwork and not play the thumping audio though.

**Comment by pchote** [comment]
No it won't.  I left it implicit that these secondary things would also be switched by upgrades.  This is exactly the same situation as the GI in RA2 (which doesn't use a transform), or at a slight stretch the MAD tank in RA.

**Comment by pchote** [comment]
My d2k-raceart branch will introduce sequence-switching on upgrade, which you will be able to use here.

**Comment by abcdefg30** [comment]
Needs a rebase again.

**Comment by Mailaender** [comment]
I wonder if these should work like http://half-life.wikia.com/wiki/Restrictor instead.

**Comment by pchote** [comment]
That could be implemented by setting a negative noise value on the attractiveness trait, but would directly contradict the dune lore.

**Comment by Phrohdoh** [comment]
@Mailaender this needs a rebase still, if you are too busy someone else may be able to take this over.

**Comment by Mailaender** [comment]
I have no plans to redo this from scratch without `Transforms` so let's stop right here.

**Comment by Mailaender** [comment]
I redid this from scratch in https://github.com/OpenRA/OpenRA/pull/9019.

**Pull request closed without merging**